apps/arweave/src/ar_node_worker.erl

```diff
@@ -1885,6 +1888,89 @@ dump_mempool(TXs, MempoolSize) ->
 			?LOG_ERROR([{event, failed_to_dump_mempool}, {reason, Reason}])
 	end.
 
+pick_prev_block_for_solution(Solution, Source) ->
+	FrontBlocks = ar_block_cache:get_validated_front(block_cache),
+	PrevBlocks = [ar_block_cache:get(block_cache, B#block.previous_block)
+			|| B <- FrontBlocks],
+	pick_prev_block_for_solution2(Solution, Source, FrontBlocks, PrevBlocks).
+
+pick_prev_block_for_solution2(Solution, Source, FrontBlocks, PrevBlocks) ->
+	#mining_solution{ mining_address = MiningAddress } = Solution,
+	case [B || B <- FrontBlocks, B#block.reward_addr == MiningAddress] of
+		[PrevB | _] ->
+			pick_prev_block_for_solution(Solution, Source, PrevB);
```

Comment:
This clause is to ensure we always build on one of our previously mined blocks if possible, right?

Reply:
Yes, we have to build on our block at this height, if any, otherwise we will double-sign.